src/style-spec/validate/validate_source.js

```diff
@@ -40,7 +40,7 @@ module.exports = function validateSource(options) {
         errors = errors.concat(validateObject({
```

Comment:
I think we can actually get rid of this case now and add `raster-dem` into the previous case with `vector` and `raster` sources. you will have to edit line 26 to something like 
```js
valueSpec: styleSpec[`source_${type.replace('-','_'}`],
```
to get it to work.

Reply:
Good point, that reduces code duplication.